riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/TestRiptideRegistrar.java

```diff
@@ -0,0 +1,55 @@
+package org.zalando.riptide.spring;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.http.client.AsyncClientHttpRequestFactory;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.web.client.AsyncRestTemplate;
+
+import static org.springframework.beans.factory.support.BeanDefinitionBuilder.genericBeanDefinition;
+
+@Slf4j
+@AllArgsConstructor
+class TestRiptideRegistrar implements RiptideRegistrar {
+
+    private final Registry registry;
+    private final RiptideSettings settings;
+
+    @Override
+    public void register() {
+        settings.getClients().forEach((id, client) -> {
+            final String templateId = registerAsyncRestTemplate(id);
+            final String serverId = registerMockRestServiceServer(id, templateId);
+            registerAsyncClientHttpRequestFactory(id, templateId, serverId);
+        });
+    }
+
+    private void registerAsyncClientHttpRequestFactory(final String id, final String templateId, final String serverId) {
+        registry.registerIfAbsent(id, AsyncClientHttpRequestFactory.class, () -> {
+            log.debug("Client [{}]: Registering mocked AsyncClientHttpRequestFactory", id);
+            final BeanDefinitionBuilder factory = genericBeanDefinition(AsyncClientHttpRequestFactory.class);
+            factory.addDependsOn(serverId);
+            factory.setFactoryMethodOnBean("getAsyncRequestFactory", templateId);
+            return factory;
+        });
+    }
+
+    private String registerAsyncRestTemplate(final String id) {
+        return registry.registerIfAbsent(id, AsyncRestTemplate.class, () -> {
```

Comment:
That will incorrectly replace the `AsyncRestTemplate` that we would register in the `DefaultRegistrar`. It would miss plugins as well as the base url.

Reply:
Fixed